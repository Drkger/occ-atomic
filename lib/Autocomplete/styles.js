import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
