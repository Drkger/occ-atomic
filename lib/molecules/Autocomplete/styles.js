export default {
    autoComplete: {
        position:'relative'
    },
    droplist: {
        position:'absolute',
        zIndex:'2',
        top:'49px',
        left:'0',
        width:'100%'
    },
    labels: {
        marginTop:'-18px',
        width:'100%'
    }
};