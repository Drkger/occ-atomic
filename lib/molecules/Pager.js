import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Icon from '../elements/Icon';
import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';

// Page

const pageViewStyles = {
    li: {
        display:'inline-block',
        cursor:'pointer',
        padding:'2px 11px',
        marginLeft:'4px',
        marginRight:'4px',
        borderRadius:'13px',
        cursor:'pointer',
        transition:'0.3s all',
        outline:'0',
        '&:hover': {
            background:'#ECECEE'
        }
    },
    active: {
        background:`${colors.blue} !important`,
        color:colors.white
    }
};

const Page = ({ classes, onClick, selected, page }) => (
    <li className={`${classes.li}${selected ? ` ${classes.active}` : ''}`} tabIndex="0" onClick={onClick}>
        {page}
    </li>
);

Page.propTypes = {
    classes: PropTypes.object,
    onClick: PropTypes.func,
    selected: PropTypes.bool,
    page: PropTypes.number
};

const PageView = injectSheet(pageViewStyles)(Page);

// Break

const breakViewStyles = {
    li: {
        display:'inline-block'
    }
};

const Break = ({ classes, label }) => (
    <li className={classes.li}>
        {label}
    </li>
);

Break.propTypes = {
    classes: PropTypes.object,
    label: PropTypes.string
};

const BreakView = injectSheet(breakViewStyles)(Break);

// Pager

const styles = {
    pager: {
        listStyle:'none',
        paddingLeft:'0',
        fontFamily:fonts.body,
        color:colors.grey7,
        fontSize:'14px'
    },
    btn: {
        display:'inline-block',
        fontFamily:fonts.body,
        border:`1px solid ${colors.grey1}`,
        borderRadius:'5px',
        padding:'3px 13px',
        transition:'0.3s all',
        cursor:'pointer',
        outline:'0',
        '& span': {
            marginBottom:'-2px'
        },
        '&:hover': {
            background:'#ECECEE',
            borderColor:'#ECECEE'
        }
    },
    prev: {
        marginRight:'15px',
        '& span': {
            transform:'rotate(90deg)'
        }
    },
    next: {
        marginLeft:'15px',
        '& span': {
            transform:'rotate(-90deg)'
        }
    },
    disabled: {
        pointerEvents:'none',
        opacity:'0.4'
    }
};

class Pager extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            selected:props.initialPage ? props.initialPage : props.forcePage ? props.forcePage : 0
        };
        this.callCallback = this.callCallback.bind(this);
        this.handlePrevPage = this.handlePrevPage.bind(this);
        this.handleNextPage = this.handleNextPage.bind(this);
        this.handlePageSelected = this.handlePageSelected.bind(this);
        this.getPageElement = this.getPageElement.bind(this);
        this.pagination = this.pagination.bind(this);
    }
    
    componentDidMount() {
        const { initialPage, disableInitialCallback } = this.props;
        if (typeof(initialPage) != 'undefined' && !disableInitialCallback) {
            this.callCallback(initialPage);
        }
    }
    
    componentWillReceiveProps(props) {
        if (typeof(props.forcePage) != 'undefined' && this.props.forcePage != props.forcePage) {
          this.setState({selected: props.forcePage});
        }
    }
    
    callCallback(selectedItem) {
        const { onPageChange } = this.props;
        if (typeof(onPageChange) != 'undefined' && typeof(onPageChange) == 'function') {
            onPageChange({selected: selectedItem});
        }
    }
    
    handlePrevPage(e) {
        const { selected } = this.state;
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (selected > 0) {
            this.handlePageSelected(selected -1, e);
        }
    }
    
    handleNextPage(e) {
        const { selected } = this.state;
        const { pageCount } = this.props;
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (selected < pageCount - 1) {
            this.handlePageSelected(selected + 1, e);
        }
    }
    
    handlePageSelected(selected, e) {
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (this.state.selected == selected) return;
        this.setState({selected});
        this.callCallback(selected);
    }
    
    getPageElement(index) {
        const { selected } = this.state;
        return (
            <PageView
                key={index}
                onClick={this.handlePageSelected.bind(null, index)}
                selected={selected == index}
                page={index+1}
            />
        );
    }
    
    pagination() {
        const items = [];
        const { selected } = this.state;
        const { pageRangeDisplayed, pageCount, marginPagesDisplayed, breakLabel } = this.props;
        
        if (pageCount <= pageRangeDisplayed) {
            for (let index = 0; index < pageCount; index++) {
                items.push(this.getPageElement(index));
            }
        } else {
            let leftSide  = (pageRangeDisplayed / 2);
            let rightSide = (pageRangeDisplayed - leftSide);
            
            if (selected > pageCount - pageRangeDisplayed / 2) {
                rightSide = pageCount - selected;
                leftSide  = pageRangeDisplayed - rightSide;
            } else if (selected < pageRangeDisplayed / 2) {
                leftSide  = selected;
                rightSide = pageRangeDisplayed - leftSide;
            }
            
            let index;
            let page;
            let breakView;
            let createPageView = (index) => this.getPageElement(index);
            
            for (index = 0; index < pageCount; index++) {
                page = index+1;
                
                // 
                if (page <= marginPagesDisplayed) {
                    items.push(createPageView(index));
                    continue;
                }
                
                if (page > pageCount - marginPagesDisplayed) {
                    items.push(createPageView(index));
                    continue;
                }
                
                if ((index >= selected - leftSide) && (index <= selected + rightSide)) {
                    items.push(createPageView(index));
                    continue;
                }
                
                if (breakLabel && items[items.length - 1] != breakView) {
                    breakView = (
                        <BreakView
                            key={index}
                            label={breakLabel}
                        />
                    );
                    items.push(breakView);
                }
            }
        }
        
        return items;
    }
    
    render() {
        const { selected } = this.state;
        const { classes, pageCount, previousLabel, nextLabel, hideNumbers } = this.props;
        return (
            <ul className={classes.pager}>
                <li className={`${classes.btn} ${classes.prev}${selected == 0 ? ` ${classes.disabled}` : ''}`} tabIndex="0" onClick={this.handlePrevPage}>
                    <Icon iconName="arrowDown" colors={['#adb4bb']} /> {previousLabel}
                </li>
                {!hideNumbers && this.pagination()}
                <li className={`${classes.btn} ${classes.next}${selected == pageCount - 1 ? ` ${classes.disabled}` : ''}`} tabIndex="0" onClick={this.handleNextPage}>
                    {nextLabel} <Icon iconName="arrowDown" colors={['#adb4bb']} />
                </li>
            </ul>
        );
    }
}

Pager.propTypes = {
    classes: PropTypes.object,
    pageCount: PropTypes.number.isRequired,
    pageRangeDisplayed: PropTypes.number.isRequired,
    marginPagesDisplayed: PropTypes.number.isRequired,
    previousLabel: PropTypes.node,
    nextLabel: PropTypes.node,
    onPageChange: PropTypes.func,
    initialPage: PropTypes.number,
    forcePage: PropTypes.number,
    disableInitialCallback: PropTypes.bool,
    hideNumbers: PropTypes.bool,
    breakLabel: PropTypes.string
};

Pager.defaultProps = {
    pageCount: 10,
    pageRangeDisplayed: 2,
    marginPagesDisplayed: 3,
    previousLabel: 'Previous',
    nextLabel: 'Next',
    breakLabel: '...',
    disableInitialCallback: false,
    hideNumbers: false
};

export default injectSheet(styles)(Pager);