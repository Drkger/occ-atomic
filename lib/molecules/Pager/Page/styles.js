import colors from '../../../subatomic/colors';

export default {
    li: {
        display:'inline-block',
        cursor:'pointer',
        padding:'2px 11px',
        marginLeft:'4px',
        marginRight:'4px',
        borderRadius:'13px',
        cursor:'pointer',
        transition:'0.3s all',
        outline:'0',
        '&:hover': {
            background:'#ECECEE'
        }
    },
    active: {
        background:`${colors.blue} !important`,
        color:colors.white
    }
};