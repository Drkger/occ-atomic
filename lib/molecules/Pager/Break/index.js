import injectSheet from 'react-jss';
import Break from './Break';
import styles from './styles';

export default injectSheet(styles)(Break);