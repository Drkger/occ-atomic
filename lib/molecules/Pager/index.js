import injectSheet from 'react-jss';
import Pager from './Pager';
import styles from './styles';

export default injectSheet(styles)(Pager);