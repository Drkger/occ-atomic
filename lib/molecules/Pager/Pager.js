import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Icon from '../../elements/Icon';

import Page from './Page';
import Break from './Break';

class Pager extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            selected:props.initialPage ? props.initialPage : props.forcePage ? props.forcePage : 0
        };
        this.callCallback = this.callCallback.bind(this);
        this.handlePrevPage = this.handlePrevPage.bind(this);
        this.handleNextPage = this.handleNextPage.bind(this);
        this.handlePageSelected = this.handlePageSelected.bind(this);
        this.getPageElement = this.getPageElement.bind(this);
        this.pagination = this.pagination.bind(this);
    }
    
    componentDidMount() {
        const { initialPage, disableInitialCallback } = this.props;
        if (typeof(initialPage) != 'undefined' && !disableInitialCallback) {
            this.callCallback(initialPage);
        }
    }
    
    componentWillReceiveProps(props) {
        if (typeof(props.forcePage) != 'undefined' && this.props.forcePage != props.forcePage) {
          this.setState({selected: props.forcePage});
        }
    }
    
    callCallback(selectedItem) {
        const { onPageChange } = this.props;
        if (typeof(onPageChange) != 'undefined' && typeof(onPageChange) == 'function') {
            onPageChange({selected: selectedItem});
        }
    }
    
    handlePrevPage(e) {
        const { selected } = this.state;
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (selected > 0) {
            this.handlePageSelected(selected -1, e);
        }
    }
    
    handleNextPage(e) {
        const { selected } = this.state;
        const { pageCount } = this.props;
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (selected < pageCount - 1) {
            this.handlePageSelected(selected + 1, e);
        }
    }
    
    handlePageSelected(selected, e) {
        e.preventDefault ? e.preventDefault() : e.returnValue = false;
        if (this.state.selected == selected) return;
        this.setState({selected});
        this.callCallback(selected);
    }
    
    getPageElement(index) {
        const { selected } = this.state;
        return (
            <Page
                key={index}
                onClick={this.handlePageSelected.bind(null, index)}
                selected={selected == index}
                page={index+1}
            />
        );
    }
    
    pagination() {
        const items = [];
        const { selected } = this.state;
        const { pageRangeDisplayed, pageCount, marginPagesDisplayed, breakLabel } = this.props;
        
        if (pageCount <= pageRangeDisplayed) {
            for (let index = 0; index < pageCount; index++) {
                items.push(this.getPageElement(index));
            }
        } else {
            let leftSide  = (pageRangeDisplayed / 2);
            let rightSide = (pageRangeDisplayed - leftSide);
            
            if (selected > pageCount - pageRangeDisplayed / 2) {
                rightSide = pageCount - selected;
                leftSide  = pageRangeDisplayed - rightSide;
            } else if (selected < pageRangeDisplayed / 2) {
                leftSide  = selected;
                rightSide = pageRangeDisplayed - leftSide;
            }
            
            let index;
            let page;
            let breakView;
            let createPage = (index) => this.getPageElement(index);
            
            for (index = 0; index < pageCount; index++) {
                page = index+1;
                
                // 
                if (page <= marginPagesDisplayed) {
                    items.push(createPage(index));
                    continue;
                }
                
                if (page > pageCount - marginPagesDisplayed) {
                    items.push(createPage(index));
                    continue;
                }
                
                if ((index >= selected - leftSide) && (index <= selected + rightSide)) {
                    items.push(createPage(index));
                    continue;
                }
                
                if (breakLabel && items[items.length - 1] != breakView) {
                    breakView = (
                        <Break
                            key={index}
                            label={breakLabel}
                        />
                    );
                    items.push(breakView);
                }
            }
        }
        
        return items;
    }
    
    render() {
        const { selected } = this.state;
        const { classes, pageCount, previousLabel, nextLabel, hideNumbers, className } = this.props;
        return (
            <ul className={`${classes.pager}${className ? ` ${className}` : ''}`}>
                <li className={`${classes.btn} ${classes.prev}${selected == 0 ? ` ${classes.disabled}` : ''}`} tabIndex="0" onClick={this.handlePrevPage}>
                    <Icon iconName="arrowDown" colors={['#adb4bb']} /> {previousLabel}
                </li>
                {!hideNumbers && this.pagination()}
                <li className={`${classes.btn} ${classes.next}${selected == pageCount - 1 ? ` ${classes.disabled}` : ''}`} tabIndex="0" onClick={this.handleNextPage}>
                    {nextLabel} <Icon iconName="arrowDown" colors={['#adb4bb']} />
                </li>
            </ul>
        );
    }
}

Pager.propTypes = {
    classes: PropTypes.object,
    className: PropTypes.string,
    pageCount: PropTypes.number.isRequired,
    pageRangeDisplayed: PropTypes.number.isRequired,
    marginPagesDisplayed: PropTypes.number.isRequired,
    previousLabel: PropTypes.node,
    nextLabel: PropTypes.node,
    onPageChange: PropTypes.func,
    initialPage: PropTypes.number,
    forcePage: PropTypes.number,
    disableInitialCallback: PropTypes.bool,
    hideNumbers: PropTypes.bool,
    breakLabel: PropTypes.string
};

Pager.defaultProps = {
    pageCount: 10,
    pageRangeDisplayed: 2,
    marginPagesDisplayed: 3,
    previousLabel: 'Previous',
    nextLabel: 'Next',
    breakLabel: '...',
    disableInitialCallback: false,
    hideNumbers: false
};

export default Pager;