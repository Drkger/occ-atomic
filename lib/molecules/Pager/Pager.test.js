import React from 'react';
import { shallow } from 'enzyme';
import Pager from './Pager';
import styles from './styles';
import PagerJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Pager", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Pager classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("PagerJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <PagerJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});