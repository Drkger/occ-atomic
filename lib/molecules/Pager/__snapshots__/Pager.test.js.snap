// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pager
    breakLabel="..."
    classes={
      Object {
        "btn": "btn",
        "disabled": "disabled",
        "next": "next",
        "pager": "pager",
        "prev": "prev",
      }
    }
    disableInitialCallback={false}
    hideNumbers={false}
    marginPagesDisplayed={3}
    nextLabel="Next"
    pageCount={10}
    pageRangeDisplayed={2}
    previousLabel="Previous"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="btn prev disabled"
          onClick={[Function]}
          tabIndex="0"
        >
          <Jss(Icon)
            colors={
              Array [
                "#adb4bb",
              ]
            }
            iconName="arrowDown"
          />
           
          Previous
        </li>,
        Array [
          <Jss(Page)
            onClick={[Function]}
            page={1}
            selected={true}
          />,
          <Jss(Page)
            onClick={[Function]}
            page={2}
            selected={false}
          />,
          <Jss(Page)
            onClick={[Function]}
            page={3}
            selected={false}
          />,
          <Jss(Break)
            label="..."
          />,
          <Jss(Page)
            onClick={[Function]}
            page={8}
            selected={false}
          />,
          <Jss(Page)
            onClick={[Function]}
            page={9}
            selected={false}
          />,
          <Jss(Page)
            onClick={[Function]}
            page={10}
            selected={false}
          />,
        ],
        <li
          className="btn next"
          onClick={[Function]}
          tabIndex="0"
        >
          Next
           
          <Jss(Icon)
            colors={
              Array [
                "#adb4bb",
              ]
            }
            iconName="arrowDown"
          />
        </li>,
      ],
      "className": "pager",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Jss(Icon)
              colors={
                Array [
                  "#adb4bb",
                ]
              }
              iconName="arrowDown"
            />,
            " ",
            "Previous",
          ],
          "className": "btn prev disabled",
          "onClick": [Function],
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "colors": Array [
                "#adb4bb",
              ],
              "iconName": "arrowDown",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          "Previous",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 1,
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 2,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 3,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "label": "...",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 8,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 9,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "page": 10,
          "selected": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Next",
            " ",
            <Jss(Icon)
              colors={
                Array [
                  "#adb4bb",
                ]
              }
              iconName="arrowDown"
            />,
          ],
          "className": "btn next",
          "onClick": [Function],
          "tabIndex": "0",
        },
        "ref": null,
        "rendered": Array [
          "Next",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "colors": Array [
                "#adb4bb",
              ],
              "iconName": "arrowDown",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="btn prev disabled"
            onClick={[Function]}
            tabIndex="0"
          >
            <Jss(Icon)
              colors={
                Array [
                  "#adb4bb",
                ]
              }
              iconName="arrowDown"
            />
             
            Previous
          </li>,
          Array [
            <Jss(Page)
              onClick={[Function]}
              page={1}
              selected={true}
            />,
            <Jss(Page)
              onClick={[Function]}
              page={2}
              selected={false}
            />,
            <Jss(Page)
              onClick={[Function]}
              page={3}
              selected={false}
            />,
            <Jss(Break)
              label="..."
            />,
            <Jss(Page)
              onClick={[Function]}
              page={8}
              selected={false}
            />,
            <Jss(Page)
              onClick={[Function]}
              page={9}
              selected={false}
            />,
            <Jss(Page)
              onClick={[Function]}
              page={10}
              selected={false}
            />,
          ],
          <li
            className="btn next"
            onClick={[Function]}
            tabIndex="0"
          >
            Next
             
            <Jss(Icon)
              colors={
                Array [
                  "#adb4bb",
                ]
              }
              iconName="arrowDown"
            />
          </li>,
        ],
        "className": "pager",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Jss(Icon)
                colors={
                  Array [
                    "#adb4bb",
                  ]
                }
                iconName="arrowDown"
              />,
              " ",
              "Previous",
            ],
            "className": "btn prev disabled",
            "onClick": [Function],
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "colors": Array [
                  "#adb4bb",
                ],
                "iconName": "arrowDown",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "Previous",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 1,
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 2,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 3,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "label": "...",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 8,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 9,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "page": 10,
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Next",
              " ",
              <Jss(Icon)
                colors={
                  Array [
                    "#adb4bb",
                  ]
                }
                iconName="arrowDown"
              />,
            ],
            "className": "btn next",
            "onClick": [Function],
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": Array [
            "Next",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "colors": Array [
                  "#adb4bb",
                ],
                "iconName": "arrowDown",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PagerJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Pager)
    breakLabel="..."
    disableInitialCallback={false}
    hideNumbers={false}
    marginPagesDisplayed={3}
    nextLabel="Next"
    pageCount={10}
    pageRangeDisplayed={2}
    previousLabel="Previous"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "breakLabel": "...",
      "classes": Object {
        "btn": "Pager-btn-0-1-2",
        "disabled": "Pager-disabled-0-1-5",
        "next": "Pager-next-0-1-4",
        "pager": "Pager-pager-0-1-1",
        "prev": "Pager-prev-0-1-3",
      },
      "disableInitialCallback": false,
      "hideNumbers": false,
      "marginPagesDisplayed": 3,
      "nextLabel": "Next",
      "pageCount": 10,
      "pageRangeDisplayed": 2,
      "previousLabel": "Previous",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "breakLabel": "...",
        "classes": Object {
          "btn": "Pager-btn-0-1-2",
          "disabled": "Pager-disabled-0-1-5",
          "next": "Pager-next-0-1-4",
          "pager": "Pager-pager-0-1-1",
          "prev": "Pager-prev-0-1-3",
        },
        "disableInitialCallback": false,
        "hideNumbers": false,
        "marginPagesDisplayed": 3,
        "nextLabel": "Next",
        "pageCount": 10,
        "pageRangeDisplayed": 2,
        "previousLabel": "Previous",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
